src/node/factory.ts

```diff
@@ -14,6 +14,133 @@
+import { TurboAuthenticatedClient } from '../common/turbo.js';
+import {
+  TokenType,
+  TurboAuthenticatedConfiguration,
+  TurboSigner,
+  TurboWallet,
+  isEthPrivateKey,
+  isJWK,
+} from '../types.js';
+import { isWeb } from '../utils/common.js';
+import { TurboWebArweaveSigner } from '../web/signer.js';
+import { TurboNodeSigner } from './signer.js';
+import { TurboAuthenticatedUploadService } from './upload.js';
+
+export class TurboFactory extends TurboBaseFactory {
+  protected static getSigner(
+    providedSigner: TurboSigner | undefined,
+    providedPrivateKey: TurboWallet | undefined,
+    token: TokenType,
+  ): TurboDataItemAbstractSigner {
+    let signer: TurboSigner;
+
+    if (providedSigner !== undefined) {
+      signer = providedSigner;
+    } else if (providedPrivateKey !== undefined) {
+      if (token === 'solana') {
+        signer = new HexSolanaSigner(providedPrivateKey);
+      } else if (token === 'ethereum') {
+        if (!isEthPrivateKey(providedPrivateKey)) {
+          throw new Error(
+            'An Ethereum private key must be provided for EthereumSigner.',
+          );
+        }
+        signer = new EthereumSigner(providedPrivateKey);
+      } else {
+        if (!isJWK(providedPrivateKey)) {
+          throw new Error('A JWK must be provided for ArweaveSigner.');
+        }
+        signer = new ArweaveSigner(providedPrivateKey as JWKInterface);
+      }
+    } else {
+      throw new Error('A privateKey or signer must be provided.');
+    }
+
+    if (isWeb()) {
+      return new TurboWebArweaveSigner({
+        signer,
+        logger: this.logger,
+      });
+    }
+    return new TurboNodeSigner({
+      signer,
+      logger: this.logger,
+    });
+  }
+
+  static authenticated({
+    privateKey,
+    signer: providedSigner,
+    paymentServiceConfig = {},
+    uploadServiceConfig = {},
+    token,
+    tokenMap,
+    gatewayUrl,
+    tokenTools,
+  }: TurboAuthenticatedConfiguration) {
+    if (!token) {
+      if (providedSigner) {
+        // Derive token from signer if not provided
+        if (providedSigner instanceof EthereumSigner) {
+          token = 'ethereum';
+        } else if (providedSigner instanceof HexSolanaSigner) {
+          token = 'solana';
+        } else {
+          token = 'arweave';
+        }
+      } else {
+        token = 'arweave';
+      }
+    }
+
+    const turboSigner = this.getSigner(providedSigner, privateKey, token);
+
+    if (!tokenTools) {
+      if (tokenMap && token === 'arweave') {
+        tokenTools = tokenMap.arweave;
+      }
+      tokenTools = defaultTokenMap[token]?.({
+        gatewayUrl,
+        logger: this.logger,
+      });
+    }
 
-export class TurboFactory extends TurboBaseFactory {}
+    const paymentService = new TurboAuthenticatedPaymentService({
+      ...paymentServiceConfig,
+      signer: turboSigner,
+      logger: this.logger,
+      token,
+      tokenTools,
+    });
+    const uploadService = isWeb()
```

Comment:
since we're in the `node` export here - we shouldn't need to worry about `isWeb` the clients will have to decide if the node export is what they want

Reply:
the reason this is here is I've witnessed the web import coming into the node directory before. but yeah I suppose it's more of a dev error in that case

Reply:
[a6c073b](https://github.com/ardriveapp/turbo-sdk/pull/126/commits/a6c073bb8b8e8bebf818f08b2ae4b4613ae28c9f)